pkg/controller/deployment/credentials_request.go

```diff
@@ -0,0 +1,147 @@
+	boundSAPath            = "token"
+	boundSAExpirySec       = 3600
+)
+
+// currently supported cloud platforms for ambient credentialsare: AWS, GCP
+var cloudCredentialConfigs = map[configv1.PlatformType]cloudCredentialsConfig{
+	configv1.AWSPlatformType: {
+		mountDirectory: awsCredentialsDir,
+		mountFilename:  awsCredentialsFileName,
+
+		secretKey: awsCredentialsSecretKey,
+	},
+	configv1.GCPPlatformType: {
+		mountDirectory: gcpCredentialsDir,
+		mountFilename:  gcpCredentialsFileName,
+
+		secretKey: gcpCredentialsSecretKey,
+	},
+}
+
+func withCloudCredentials(secretsInformer coreinformersv1.SecretInformer, infraInformer configinformersv1.InfrastructureInformer, deploymentName, secretName string) func(operatorSpec *v1.OperatorSpec, deployment *appsv1.Deployment) error {
+	// cloud credentials is only required for the controller deployment,
+	// other deployments should be left untouched
+	if deploymentName != "cert-manager" {
+		return func(operatorSpec *v1.OperatorSpec, deployment *appsv1.Deployment) error {
+			return nil
+		}
+	}
+
+	return func(operatorSpec *v1.OperatorSpec, deployment *appsv1.Deployment) error {
+		volumes := []corev1.Volume{{
+			Name: boundSATokenVolumeName,
+			VolumeSource: corev1.VolumeSource{
+				Projected: &corev1.ProjectedVolumeSource{
+					DefaultMode: pointer.Int32(420),
+					Sources: []corev1.VolumeProjection{{
+						ServiceAccountToken: &corev1.ServiceAccountTokenProjection{
+							Audience:          boundSAAudience,
+							ExpirationSeconds: pointer.Int64(boundSAExpirySec),
+							Path:              boundSAPath,
+						}},
+					},
+				},
+			},
+		}}
+		volumeMounts := []corev1.VolumeMount{{
+			Name:      boundSATokenVolumeName,
+			MountPath: boundSATokenDir,
+			ReadOnly:  true,
+		}}
+
+		if len(secretName) > 0 {
+			_, err := secretsInformer.Lister().Secrets(operatorclient.TargetNamespace).Get(secretName)
+			if err != nil {
+				return err
+			}
+
+			if err != nil && apierrors.IsNotFound(err) {
+				return fmt.Errorf("(Retrying) cloud secret %q doesn't exist due to %v", secretName, err)
+			} else if err != nil {
+				return err
+			}
+
+			infra, err := infraInformer.Lister().Get("cluster")
+			if err != nil {
+				return err
+			}
+
+			cloudProvider := infra.Status.PlatformStatus.Type
+			if cloudCredentialConfig, ok := cloudCredentialConfigs[cloudProvider]; ok {
+				// supported cloud platform for mounting secrets found
+				volumes = append(volumes, corev1.Volume{
+					Name: cloudCredentialsVolumeName,
+					VolumeSource: corev1.VolumeSource{
+						Secret: &corev1.SecretVolumeSource{
+							SecretName: secretName,
+							Items: []corev1.KeyToPath{{
+								Key:  cloudCredentialConfig.secretKey,
+								Path: cloudCredentialConfig.mountFilename,
+							}},
+						},
+					},
+				})
+				volumeMounts = append(volumeMounts, corev1.VolumeMount{
+					Name:      cloudCredentialsVolumeName,
+					MountPath: cloudCredentialConfig.mountDirectory,
+				})
+			}
```

Comment:
Can you change this to a switch and drop `cloudCredentialConfigs`? I think it would make it more readable and maintainable.

Reply:
Refactored to use switch case instead.